enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,399 @@
+---
+title: bpfd-on-ocp
+authors:
+  - "@anfredette"
+  - "@astoycos"
+  - "@Billy99"
+  - "@dave-tucker"
+reviewers:
+  - "@TBD"
+approvers:
+  - "@TBD"
+api-approvers: 
+  - "@TBD"
+creation-date: 2023-02-21
+last-updated: 2023-02-21
+tracking-link:
+  # Tracking link TBD, but need a valid link here here to pass linter.
+  - "https://bpfd.netlify.app"
+see-also:
+replaces:
+superseded-by:
+---
+
+# BPF Support for OCP with bpfd
+
+## Summary
+
+This enhancement request proposes the following:
+- That the bpfd operator be included in OCP as a foundational operator.
```

Comment:
We are currently trying to evolve OCP so that more features like this are delivered as add-on operators, rather than as part of the release payload. The bar for adding new things to the payload is basically that it must be present to form the cluster or a significant number of clusters (nearly all of them) will use the feature.

Is it possible to deliver the features described in this enhancement via an add-on operator?

Reply:
The thought on this one is that every Red Hat feature that uses BPF would use bpfd to load and manage its BPF programs.  These currently include Ingress Node Firewall, Network Observability, and ACS, and the list is expected to grow.  If this proposal is accepted, then we were thinking that it would be necessary for bpfd to be always available.

Reply:
I believe this has to be a core feature. Our long-term networking strategy includes making use of eBPF in the networking data plane. That means that this is a requirement for core networking features. This has to be in place first before that can proceed. In component terms, you could see this as a dependency of the cluster-network-operator.

Reply:
Which team or teams within the OCP org would be responsible for maintaining bpfd?